core/src/main/java/hudson/tools/DownloadFromUrlInstaller.java

```diff
@@ -126,7 +127,61 @@ protected DescriptorImpl() {
             Downloadable.all().add(createDownloadable());
         }
 
-        protected Downloadable createDownloadable() {
+        /**
+         * function that creates a {@link Downloadable}.
+         * @return a downloadable object
+         */
+        public Downloadable createDownloadable() {
+            if (getId().equals("hudson.tasks.Maven.MavenInstaller") || getId().equals("hudson.tasks.Ant.AntInstaller")) {
```

Comment:
I realize that I've just lobbed another [wave of mutilation](https://www.youtube.com/watch?v=hvHJOus1pRo) at your pull request ;-)